src/modules/imports/import.rs

```diff
@@ -146,20 +146,24 @@ impl SyntaxModule<ParserMetadata> for Import {
                 token(meta, "{")?;
                 let mut exports = vec![];
                 loop {
+                    if token(meta, "}").is_ok() {
```

Comment:
This token check is only required on the first iteration through the loop, to catch an empty import list. I believe that on all subsequent iterations, any trailing brace is caught by the other check below.

It won't make any noticeable difference to the performance, but it may help subsequent developers understand the logic better, if you move the token check out of the loop.  You might also change `vec![]` to `Vec::new()`:

```
token(meta, "{")?;
let mut exports = Vec::new();
if token(meta, "}").is_err() {
    loop {
        let tok = meta.get_current_token();
        let name = variable(meta, variable_name_extensions())?;
...
```

Reply:
> You might also change `vec![]` to `Vec::new()`

why? we use `vec![]` throughout the codebase. there is no reason to have an inconsistency like this

Reply:
Okay, fair enough. In isolated code, I would prefer a straight `Vec::new()` call because simpler is better, and you really only need `vec![]` to initialise an array from a sequence of objects. However, I'm all for internal consistency in a project; happy for it to remain as-is.